### PR TITLE
t/330: Introduce AttributeElement and ContainerElement.

### PR DESCRIPTION
**Pull request by pjasiun**
Fixes: #330.

### CODE REVIEW

File: src/treeview/element.js [hunk 1]
```diff
@@ -110,7 +118,8 @@ export default class Element extends Node {
 			}
 		}
 
-		const cloned = new Element( this.name, this._attrs, childrenClone );
+		// ContainerElement and AttributeElement should be also cloned properly.
+		const cloned = new this.constructor( this.name, this._attrs, childrenClone );
```
Comment (szymonkups) [inline src/treeview/element.js]: We should also clone `AttributeElement`'s priority.

File: src/treeview/writer.js [hunk 2]
```diff
@@ -414,10 +342,19 @@ import DocumentFragment from './documentfragment.js';
 	 * and {@link engine.treeView.Range#end} positions are not placed inside same parent container.
 	 *
 	 * @param {engine.treeView.Range} range Range to wrap.
-	 * @param {engine.treeView.Element} attribute Attribute element to use as wrapper.
+	 * @param {engine.treeView.AttributeElement} attribute Attribute element to use as wrapper.
 	 * @param {Number} priority Priority to set.
 	 */
 	wrap( range, attribute, priority ) {
```
Comment (szymonkups) [inline src/treeview/writer.js]: `AttributeElement` should store priority so there is no need now to pass it as third parameter.

Comment (szymonkups) [inline src/treeview/writer.js]: `AttributeElement` store priority itself so there is no need now to pass it as third parameter.

File: src/treeview/element.js (hunk 1 above)
Comment (pjasiun) [inline src/treeview/element.js]: Fixed.

**Comment by pjasiun** [comment]
Pushed commits with all mentioned changes. Also I added `isSimmilar` method to the `AttributeElement` which additionally check if priorities are the same.

**Pull request merged**